dc/core/DisplayGroup.cpp

```diff
@@ -203,29 +204,36 @@ void DisplayGroup::unfocus( const QUuid& id )
         return;
 
     window->setFocused( false );
-    removeFocusedWindow( window );
+    _removeFocusedWindow( window );
     // Make sure the window dimensions are re-adjusted to the new zoom level
     window->getController()->scale( window->getCoordinates().center(), 0.0 );
+    _updateFocusedWindowsCoordinates();
+
     sendDisplayGroup();
 }
 
+const std::set<ContentWindowPtr>& DisplayGroup::getFocusedWindows() const
```

Comment:
I decided to keep the two different containers with a new typedef. The full set of ContentWindowPtrs has to be ordered to determine rendering order, so it can't be a set, but this is not needed for the focusedWindows.